pandas/core/dtypes/cast.py

```diff
@@ -357,15 +357,13 @@ def trans(x):
             # if we don't have any elements, just astype it
             return trans(result).astype(dtype)
 
-        # do a test on the first element, if it fails then we are done
-        r = result.ravel()
-        arr = np.array([r[0]])
-
-        if isna(arr).any():
-            # if we have any nulls, then we are done
-            return result
-
-        elif not isinstance(r[0], (np.integer, np.floating, int, float, bool)):
+        if isinstance(result, np.ndarray):
+            element = result[0]
```

Comment:
would `result.item(0)` work here? might avoid potential copies from ravel

Reply:
My suggestion here turns out to cause a different problem bc .item(0) is wrong for dt64 ndarrays

```
result = np.arange(5).view("M8[ns]")
>>> result.item(0)
0
```